src/main/java/org/entando/entando/aps/system/services/userpreferences/UserPreferencesService.java

```diff
@@ -30,8 +32,14 @@
 
     IUserPreferencesManager userPreferencesManager;
 
+    // Striped lock used to allow different users to access concurrently the
+    // getUserPreferences() method but forcing the same user to access it sequentially
+    private final Striped<Lock> newPreferencesLocks = Striped.lazyWeakLock(64);
```

Comment:
Creation of a new user preference if it doesn't exist should be implemented as an atomic operation. I used this class to implement a fine grain check that forces the sequential access only if the calls are related to the same user. I'm not sure about the right number of stripes to use. Maybe using this is overkill for our use case.